source/MaterialXGenShader/CMakeLists.txt

```diff
@@ -9,5 +16,14 @@ mx_add_library(MaterialXGenShader
     LIBRARIES
         MaterialXFormat
         MaterialXCore
+        ${optionalLibraries}
     EXPORT_DEFINE
         MATERIALX_GENSHADER_EXPORTS)
+
+if(MATERIALX_BUILD_OCIO)
+    if (NOT MATERIALX_BUILD_MONOLITHIC)
+        target_compile_definitions(MaterialXGenShader PUBLIC MATERIALX_BUILD_OCIO)
+    else()
+        target_compile_definitions(${MATERIALX_MONOLITHIC_TARGET} PUBLIC MATERIALX_BUILD_OCIO)
```

Comment:
You should be able to use `${TARGET_NAME}` to collect the correct monolithic/non-monolithic target to use.

It might also be cleaner to append the OCIO library to the target that way too, so then the entire OCIO addition here could just be added inside one guard

```
if(MATERIALX_BUILD_OCIO)
    find_package(OpenColorIO REQUIRED)
    target_link_libraries(${TARGET_NAME} OpenColorIO::OpenColorIO)
    target_compile_definitions(${TARGET_NAME} PUBLIC MATERIALX_BUILD_OCIO)
endif()
```